### PR TITLE
Remove observable interface

### PR DESCRIPTION
**Pull request by clostao**
This PR substitutes the observable returning type of upload methods for a returning promise and an optional callback for tracking the upload. 

Additionally, it updates the module type to `node16` for assuring that asynchronous `imports` are not transpiled to `require` statements.

### CODE REVIEW

**Review by marc-aurele-besner — CHANGES_REQUESTED** [review]
This is great! 💪 
Can we reflect these changes in the package README please?

**Comment by clostao** [comment]
Updated, thanks!

File: packages/auto-drive/README.md
Comment (marc-aurele-besner) [inline packages/auto-drive/README.md:1]: most import are missing `createAutoDriveApi`

**Review by marc-aurele-besner — COMMENTED** [review]
Awesome, just some small miss in most import in the readme, let's try to make these code snippet as copy/pastable as possible without errors 👍

**Review by marc-aurele-besner — APPROVED** [review]
Awesome work, thank you very much 🙇‍♂️

**Pull request merged**